calico/src/main/scala/calico/html.scala

```diff
@@ -499,6 +502,28 @@ trait ClassPropModifiers[F[_]](using F: Async[F]):
   private val _forConstantClassProp: Modifier[F, Any, SingleConstantModifier] =
     (m, n) => Resource.eval(F.delay(n.asInstanceOf[js.Dictionary[String]]("className") = m.cls))
 
+final class DataProp[F[_]] private[calico] (suffix: String)
+    extends Prop[F, DataProp.SingleConstantModifier, String](
+      s"data-$suffix",
```

Comment:
I'm not sure if this is defined correctly. Compare with `ClassProp`.
https://github.com/armanbilge/calico/blob/c96718ba1e5f63c54837aa9a78772059040aa71a/calico/src/main/scala/calico/html.scala#L474-L475

Reply:
> I'm not sure if this is defined correctly. Compare with `ClassProp`.

Hm, indeed I guess there's the property name needed instead of attribute name. That'd be `dataset`. Interestingly it works nonetheless in my simple use case.

Reply:
Actually, I was confused about the type parameter: `DataProp.SingleConstantModifier`.

Reply:
Ahhh, interesting! Do you have an example how it's used?

Reply:
Sure: ```
          div(
            data("bsToggle") := "collapse",
            cls := "todoapp",
```
Attribute view: `<div class="todoapp" data-bs-toggle="collapse"`. Example is nonsensical but it should show the point.

Reply:
Thanks, that's helpful! So then I think I'm confused why `DataProp` extends `Prop`, since it defines its own `:=` ?

Reply:
> Thanks, that's helpful! So then I think I'm confused why `DataProp` extends `Prop`, since it defines its own `:=` ?

That's what I meant by "quickly hacked together" ;) I think we can reuse most functionality for `DataProp` from `Prop` by just abstracting a bit over `setProp`. Will PR this tomorrow.

Reply:
Should look a bit better now

Reply:
Should look better now